tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -15756,15 +15793,17 @@ void TryAddShaderAttrFromTargetProfile(Sema &S, FunctionDecl *FD,
   }
 
   HLSLShaderAttr *currentShaderAttr = FD->getAttr<HLSLShaderAttr>();
-  // Don't add the attribute if it already exists as an attribute on the decl.
-  // In the special case that the target profile is compute and the
-  // entry decl already has a node shader attr, don't do anything
+  // Don't add the attribute if it already exists as an attribute on the decl,
+  // and emit an error. In the special case that the target profile is compute
+  // and the entry decl already has a node shader attr, don't do anything
   if (currentShaderAttr) {
     llvm::StringRef currentFullName = currentShaderAttr->getStage();
     if (currentFullName != fullName) {
+
       S.Diag(currentShaderAttr->getLocation(),
              diag::err_hlsl_profile_conflicts_with_shader_attribute)
           << fullName << profile << currentFullName << EntryPointName;
+      return;
```

Comment:
We would have returned already in the outer `if` block, so I'm not sure what difference this return makes.

Also of note, below that outer return, we set `isActiveEntry = true`.  I think this should be set before looking for a current shader attribute, since at that point, we know this is the entry point referred to by the HLSLEntryFunction setting.  Right now, if you have a shader attribute on your function that you compile through an ordinary shader target, this code will set isActiveEntry to false, which is not the desired behavior.